metadata/src/test/java/org/apache/kafka/controller/ConfigurationControlManagerTest.java

```diff
@@ -245,4 +246,32 @@ public void testLegacyAlterConfigs() {
             manager.legacyAlterConfigs(toMap(entry(MYTOPIC, toMap(entry("def", "901")))))
         );
     }
+
+    @Test
+    public void testShouldUseUncleanLeaderElection() throws Exception {
+        SnapshotRegistry snapshotRegistry = new SnapshotRegistry(new LogContext());
+        ConfigurationControlManager manager =
+            new ConfigurationControlManager(new LogContext(), 0, snapshotRegistry, CONFIGS);
+        ControllerResult<Map<ConfigResource, ApiError>> result = manager.incrementalAlterConfigs(
+            toMap(entry(BROKER0, toMap(
+                    entry(UNCLEAN_LEADER_ELECTION_ENABLE_CONFIG, entry(SET, "true")))),
+                entry(MYTOPIC, toMap(
+                    entry(UNCLEAN_LEADER_ELECTION_ENABLE_CONFIG, entry(SET, "false"))))));
+        Map<ConfigResource, ApiError> expectedResponse = new HashMap<>();
+        expectedResponse.put(BROKER0, ApiError.NONE);
+        expectedResponse.put(MYTOPIC, ApiError.NONE);
+        assertEquals(expectedResponse, result.response());
+        ControllerTestUtils.replayAll(manager, result.records());
+        assertTrue(manager.shouldUseUncleanLeaderElection(MYTOPIC.name()));
```

Comment:
I think in this case, unclean leader election should be false for MYTOPIC since the topic level config takes precedence.